### PR TITLE
use Maps to only iterate through slices that actually have a given action

### PR DESCRIPTION
**Pull request by EskiMojo14**
i saw a FIXME and got an itch

there isn't really a test suite yet as far as i can tell, but i tried it out with the example and it still worked correctly

### CODE REVIEW

**Comment by EskiMojo14** [comment]
out of curiosity, is calling `set` multiple times particularly inefficient? you could always do the iteration inside a single `set` call instead, building out the object as you go

**Comment by dai-shi** [comment]
wow, nice! thanks for working on it.
yeah, let's add some tests first. would you like to work on it? using `@testing-library/react` and testing at the component level would be good.

right, calling `set` multiple times isn't super efficient (it's still render optimized, though.)

**Comment by EskiMojo14** [comment]
sure - do you want a separate PR for tests or should I just add them to this one?

**Comment by EskiMojo14** [comment]
opened #3 with tests

File: src/index.ts [hunk 1]
```diff
@@ -91,26 +91,28 @@ export function withSlices<Configs extends SliceConfig<string, unknown, any>[]>(
...
-    const actionNameSet = new Set<string>();
+    const sliceMapsByAction = new Map<string, Map<string, any>>();
     for (const config of configs) {
       state[config.name] = config.value;
-      for (const actionName of Object.keys(config.actions)) {
-        actionNameSet.add(actionName);
+      for (const [actionName, actionFn] of Object.entries(config.actions)) {
+        let actionsBySlice = sliceMapsByAction.get(actionName);
+        if (!actionsBySlice) {
+          sliceMapsByAction.set(actionName, (actionsBySlice = new Map()));
```
Comment (dai-shi) [inline src/index.ts:100]: Oh, I like this syntax. I think I used two statements before for this pattern.

**Review by dai-shi — APPROVED** [review]
Looks good to me!

Thanks for your work.

**Pull request merged**